### PR TITLE
New package name and directory structure for `linkml-round-trips`

### LINKED ISSUE #20: general factoring
- more class orientation
- better name for `linkml_round_trips` folder, broken out into topical subdirectories

### LINKED ISSUE #65: add click log back into spatil/new-directory-structure (esp. sheet2linkml script)

### LINKED ISSUE #66: revert default `--client_secret_json` parameter for `sheet2linkml` in spatil/new-directory-structure
default `--client_secret_json` parameter for `sheet2linkml` script should be `local/client_secret.apps.googleusercontent.com.json`

### LINKED ISSUE #68: mising parent class errors in data.tsv -> data.js conversion (branch spatil/new-directory-structure)
That conversion is performed by  by `script/make_data.py`

> Error: parent class  history/tillage doesn't exist as section or field for term. Make sure parent term is trimmed of whitespace. zonal tillage

### LINKED ISSUE #69: jgi_gen:default only section in DH template (branch spatil/new-directory-structure)
dependent on arguments to linkml to DH script

### PR DESCRIPTION
**Pull request by sujaypatil96**
This PR seeks to address the first point in issue #20. It proposes a new directory structure for the `linkml-round-trips` folder, which is essentially a package within the repo, per the following:

```
sheet2dataharmonizer
├── __init__.py
└── converters
    ├── __init__.py
    ├── commands.py
    ├── linkml2dataharmonizer.py
    └── sheet2linkml.py
```

The mapping of files from the previous names to the new names is as follows:

1. `linkml_to_dh_light` : `linkml2dataharmonizer.py`
2. `modular_gd.py` : `sheet2linkml.py`
3. `use_modular_gd.py` : `__main__.py`

Note: The click logic has been moved to `commands.py`.

CC: @turbomam

### CODE REVIEW

File: sheet2dataharmonizer/__main__.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    emsl_sheet = mgd.get_gsheet_frame(client_secret_json, sheet_id, "EMSL_sample_slots")
+    emsl_dict = emsl_sheet.to_dict(orient="records")
+
+    new_schema = add_emsl_schema(emsl_dict, new_schema)
+
+    # can be executed in runtime
+    # consider moving into __name__ == "__main__"
+    dumped = yaml_dumper.dumps(new_schema)
+
+    logger.info(dumped)
```
Comment (turbomam) [inline sheet2dataharmonizer/__main__.py]: Thanks, this looks way better in most ways.

I think this should still be a print statement to STDOUT. At least that' the way the Makefile expects it now. We could have an output file argument and write to that. But I can't see how logging the dumped schema would work.

File: sheet2dataharmonizer/__main__.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            task["focus_class"],
+            pysqldf_slot_list,
+            new_schema,
+            constructed_class_name,
+        )
+
+    return new_schema
+
+
+def add_emsl_schema(emsl_dict, new_schema):
```
Comment (turbomam) [inline sheet2dataharmonizer/__main__.py]: It looks like this replaces the following:

```
    new_schema = inject_supplementary(client_secret_json, sheet_id, 'EMSL_sample_slots', new_schema, "emsl",
                                      constructed_class_name)
```

I was also applying the `inject_supplementary` function to the `mixs_modified_slots` and `biosample_identification_slots` sheets from [Soil-NMDC-Template_Compiled](https://docs.google.com/spreadsheets/d/1pSmxX6XGOxmoA7S7rKyj5OaEl3PmAl4jAOlROuNHrU0)

Are you pretty sure the revise code sill processes those?

**Comment by sujaypatil96** [comment]
@turbomam: I think we're ready to merge in the code from the first round of refactor, which primarily focussed on setting up a semantic directory structure, with the main focus being `__main__.py`. Future rounds will focus on making modules adhere to OOP design.

**Comment by sujaypatil96** [comment]
This PR merged in the changes from PR #60, which included changes to the design pattern of each of the modules within the package.

**Comment by sujaypatil96** [comment]
Apologies for the many commits after requesting a review, but I've tested it fully and I think it's ready to be beta tested and merged.

File: pyproject.toml [hunk 3]
```diff
@@ -30,10 +30,11 @@ build-backend = "poetry.core.masonry.api"
...
-#linkml_to_dh_light = "linkml_round_trips.linkml_to_dh_light:linkml_to_dh_light"
+interleave_classes = "linkml_round_trips.old.interleave_classes:interleave_classes"
+get_dependencies = "linkml_round_trips.old.get_dependencies:get_dependencies"
+merge_dont_interleave = "linkml_round_trips.old.merge_dont_interleave:merge_dont_interleave"
+interleave_mergeds = "linkml_round_trips.old.interleave_mergeds:interleave_mergeds"
+linkml_to_dh_light = "linkml_round_trips.linkml_to_dh_light:linkml_to_dh_light"
+linkml2dataharmonizer = "sheet2dataharmonizer.converters.commands:linkml2dataharmonizer"
 mixs_package_classes = "linkml_round_trips.old.mixs_package_classes:mixs_package_classes"
-combine_schemas = "use_modular_gd:combine_schemas"
+sheet2linkml = "sheet2dataharmonizer.__main__:main"
```
Comment (cmungall) [inline pyproject.toml]: minor: I like having slightly more specific names, like nmdcsheet2linkml

**Review by cmungall — APPROVED** [review]

**Comment by sujaypatil96** [comment]
@turbomam: I think I've managed to resolve issue #69. Could you verify by running the appropriate make command on your end? I managed to generate the correct `data.tsv`, but I seem to be having issues generating the HTML and CSS files.

**Comment by sujaypatil96** [comment]
@turbomam: I tested the code after all the latest commits on this PR, and everything seems to work just like the old code. Please feel free to test it, and let me know if you think it's ready to replace the old package.

**Comment by turbomam** [comment]
Trying now, Please change the default `--client_secret_json` for `sheet2linkml` back to  `client_secret.apps.googleusercontent.com.json`

**Comment by turbomam** [comment]
```shell
poetry run enum_annotator \
	--modelfile target/soil_biosample_modular.yaml \
	--requested_enum_name fao_class_enum \
	--ontology_string ENVO \
	--max_cosine 0.1  > target/temp1.yaml 2>> target/annotation.log
```

errored out for me. I just used it for another project today and it worked, but it is dependent on the EBI's OLS API.

**Comment by turbomam** [comment]
> errored out for me. I just used it for another project today and it worked, but it is dependent on the EBI's OLS API.

`enum_annotator` errored out in my synthetic biology repo now too. :-(

**Review by turbomam — APPROVED** [review]
I did this annecdotal comparison:

```shell
$ diff data_spatil_new-directory-structure.tsv data_master.tsv 
$ 
```

I think it's ready to merge

I'll probably need some hand holding re getting used to the new structure

**Pull request merged**